proxy/pkg/zdmproxy/clientconn.go

```diff
@@ -174,14 +175,15 @@ func (cc *ClientConnector) listenForRequests() {
 					err, cc.clientHandlerContext, cc.clientHandlerCancelFunc, ClientConnectorLogPrefix, "reading", connectionAddr)
 				break
 			} else if protocolErrResponseFrame != nil {
-				f = protocolErrResponseFrame
-				if !protocolErrOccurred {
-					protocolErrOccurred = true
-					cc.sendResponseToClient(protocolErrResponseFrame)
-					cc.clientHandlerShutdownRequestCancelFn()
```

Comment:
It looks like `clientHandlerShutdownRequestCancelFn` is not called anywhere anymore, we should probably remove it from the struct.

Reply:
We want to call it once #68 is addressed, maybe I can add a comment about this

Reply:
done